packages/core-data/src/selectors.ts

```diff
@@ -19,6 +19,19 @@ import { getQueriedItems } from './queried-data';
 import { DEFAULT_ENTITY_KEY } from './entities';
 import { getNormalizedCommaSeparable, isRawAttribute } from './utils';
 import type { Context, User, Theme, WpTemplate } from './entity-types';
+import {
+	DefaultContextOf,
+	EntityRecordOf,
+	KeyOf,
+	Kind,
+	KindOf,
+	Name,
+	NameOf,
+} from './entity-types';
+
+// createSelector isn't properly typed if I don't explicitly import these files – ideally they would
+// be merely ambient definitions that TS is aware of.
+import type {} from './rememo';
```

Comment:
I'm not sure what's the best way of using this ambient type without TS warnings before enabling `"types"` in package.json – these two seem to be connected.

Reply:
Looks like y'all are on older version of rememo (v3.0.0). First-party type definitions are available as of [v4.0.0](https://github.com/aduth/rememo/blob/master/CHANGELOG.md#v400-2021-07-11), so should hopefully be as simple as upgrading.

![image](https://user-images.githubusercontent.com/1779930/170792895-381f89d7-2669-4ac4-adfc-5d314e9d84bf.png)

Reply:
Looks like y'all are on older version of rememo (v3.0.0). First-party type definitions are available as of [v4.0.0](https://github.com/aduth/rememo/blob/master/CHANGELOG.md#v400-2021-07-11), so should hopefully be as simple as upgrading.

![image](https://user-images.githubusercontent.com/1779930/170792895-381f89d7-2669-4ac4-adfc-5d314e9d84bf.png)

Reply:
Since it's used across many different packages, I opened a separate pull request to upgrade them all in one go: #41415

Reply:
I went ahead and merged that PR, thank you so much @aduth! I am AFK this week so I will rebase this one in a few days to confirm it went well.

Reply:
Oh hi :)